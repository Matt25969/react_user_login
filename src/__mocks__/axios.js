
module.exports = {
  get: (url) => {
      if (url === "http://localhost:5000/user/name/valid/valid"){
    return Promise.resolve({data:{Status : "Logged In"}});
  }
  else
  {
    return Promise.resolve({data:{Status : "Not Logged In"}});
  }
  },


  post: (url, body) => {

  if (body.username==='invalid'){
  
  return Promise.resolve({data:{username:"Username field is required"}});
  }
  if (body.email==='invalid'){
  
    return Promise.resolve({data:{email:"Email field is required"}});
    }
}

  
};