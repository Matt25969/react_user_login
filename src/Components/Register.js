import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import axios from 'axios';

import '../App.css';

export default class Register extends Component {

    constructor(props) {
        super(props);
        this.state = {
            data:[],
            error:{
                username:"",
                email:""
            }
        }
    }

    handleSubmit = (e) => {

        e.preventDefault();

        let newUser = {
            username:e.target[0].value,
            email:e.target[1].value,
            password:e.target[2].value,
            password2:e.target[3].value
        }

        axios.post("http://localhost:5000/user/createuser",newUser).then(res => {


        this.setState({
            error:res.data
        })

        console.log(res)

            this.props.onLoad()
        
        }).catch(err => {
        console.log(err)
            
           
        })

    }

render() {
  return (
    <div>
        
        <form id="RegisterForm" onSubmit={this.handleSubmit}>
<input placeholder="username"></input>
<br></br>
<input placeholder="email"></input>
<br></br>
<input placeholder="password"></input>
<br></br>
<input placeholder="password2"></input>
<br></br>
<input type="submit"></input>
</form>
<p id="usernameError">{this.state.error.username}</p>
<p id="emailError">{this.state.error.email}</p>
<p>{this.state.error.password}</p>
<p>{this.state.error.password2}</p>
    </div>
  );
}
    }
