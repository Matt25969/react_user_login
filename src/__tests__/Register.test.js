import React from 'react';
import axios from 'axios';

import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16';
import {  mount, configure } from 'enzyme';

import Register from '../Components/Register'

jest.mock('axios');

configure({ adapter: new Adapter() });

it('renders Register comp', () => {
  const tree = renderer.create(<Register />).toTree;
  
  expect(tree).toMatchSnapshot();
})

it('Button submits form with invalid username', () => {

  const postSpy = jest.spyOn(axios, 'post');

  const context = {};

  const wrapper = mount(<Register />, { context });

    const e = {
        target: [{
            value: "invalid"
        },
        {
            value: "valid"
        },
        {
            value: "valid"
        },
        {
            value: "valid"
        }]
    };

    wrapper.find('[id="RegisterForm"]').last().simulate("submit", e);

    const postPromise = postSpy.mock.results.pop().value;

    return postPromise.then(() => {
      
      const currentState = wrapper.state();

      wrapper.setContext({});

      expect(wrapper.contains(<p id="usernameError">Username field is required</p>)).toBe(true);
      
      expect(currentState.error.username=='Username field is required').toBe(true);

    })

});

it('Button submits form with invalid email', () => {

  const postSpy = jest.spyOn(axios, 'post');

  const context = {};

  const wrapper = mount(<Register />, { context });

    const e = {
        target: [{
            value: "valid"
        },
        {
            value: "invalid"
        },
        {
            value: "valid"
        },
        {
            value: "valid"
        }]
    };

    wrapper.find('[id="RegisterForm"]').last().simulate("submit", e);

    const postPromise = postSpy.mock.results.pop().value;

    return postPromise.then(() => {
      
      const currentState = wrapper.state();

      wrapper.setContext({});

      expect(wrapper.contains(<p id="emailError">Email field is required</p>)).toBe(true);
      
      expect(currentState.error.email=='Email field is required').toBe(true);

    })

});

