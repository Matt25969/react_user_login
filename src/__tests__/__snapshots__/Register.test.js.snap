// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login comp 1`] = `[Function]`;

exports[`renders Register comp 1`] = `[Function]`;
